### PR TITLE
Auto add e2e label + PR description reminder for e2e

### PR DESCRIPTION
**Pull request by RamIdeas**
## What this PR solves / how to test

Sometimes we review PRs that change snapshot values in unit-tests and e2e tests but we don't notice the e2e snapshots were updated.

This PR makes sure that the 'e2e' label is added to a PR if any files within wrangler's e2e/ directory are touched. If the e2e tests change, we undoubtedly want to run the job to make sure they still pass.

This PR also adds to the PR description a reminder for the author and the reviewers to check whether an e2e test run is required.

## Author has addressed the following

- Tests
  - [ ] TODO (before merge)
  - [ ] Included
  - [x] Not necessary because: github action workflow change
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [ ] TODO (before merge)
  - [ ] Included
  - [x] Not necessary because: internal repo change
- Public documentation
  - [ ] TODO (before merge)
  - [ ] Cloudflare docs PR(s): <https://github.com/cloudflare/cloudflare-docs/pull/>...
  - [x] Not necessary because: internal repo change

### CODE REVIEW

File: .github/pull_request_template.md
Comment (RamIdeas) [inline .github/pull_request_template.md:12]: This is useful for external contributors who don't know whether their change requires an e2e run and reminds us, as reviewers, to decide for ourselves.

File: .github/pull_request_template.md
Comment (RamIdeas) [inline .github/pull_request_template.md:13]: This is useful for internal and external contributors.

For internal contributors, this will serve as a reminder to add the "e2e" label before the PR is merged. Especially when you don't want to run them while the PR is in-flight to save runners.

For external contributors, they won't be able to run the e2e tests themselves (due to security implications). This will serve as a reminder to add the "e2e" label and also for us, as maintainers, to run the e2e tests in a separate PR (only once we have reviewed the changes and confirmed no malicious intent)

File: .github/pull_request_template.md
Comment (RamIdeas) [inline .github/pull_request_template.md:14]: This is useful to confirm there is no need for the e2e test run. And to make sure they weren't run only because we forgot to add the label.

File: .github/workflows/e2e.yml
Comment (petebacondarwin) [inline .github/workflows/e2e.yml:8]: Is this enough? It feels like this actually restricts the job to run only on pushes that match both the branch and the path?

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushbranchestagsbranches-ignoretags-ignore:~:text=the%20path%20again.-,If%20you%20define%20both%20branches/branches%2Dignore%20and%20paths/paths%2Dignore%2C%20the%20workflow%20will%20only%20run%20when%20both%20filters%20are%20satisfied.,-The%20paths%20and

Comment (RamIdeas) [reply .github/workflows/e2e.yml:8]: Good catch! Any idea how to make it run if either match?

Trying to make "push:" an array so I can specify the same event twice here https://github.com/cloudflare/workers-sdk/pull/5793/commits/0daaaf008ca205bbf36b37cac03a3e60c35b8f71

Comment (RamIdeas) [reply .github/workflows/e2e.yml]: I've switched strategy. Rather than mess with the existing job, I've added a job to automatically add the 'e2e' label if files within the dir change (this feature comes out-of-the-box with actions/labeler)

**Review by petebacondarwin — APPROVED** [review]
Nice!

**Pull request merged**
